pkg/render/manager_route_config.go

```diff
@@ -0,0 +1,416 @@
+	// and validated and the route is added to the ConfigMap mounted to voltron to configure the tls terminated routes.
+	//
+	// If CAs or MTLS certs / keys are referenced in the spec the Config Maps and Secrets containing
+	// those values must be added through AddConfigMap or AddSecret. This is so we can track when these values change.
+	AddTLSTerminatedRoute(routes operatorv1.TLSTerminatedRoute)
+
+	// AddTLSPassThroughRoute adds TLSTerminatedRoutes to the config builder. When Build is called, the route is parsed
+	// and validated and the route is added to the ConfigMap mounted to voltron to configure the tls pass through routes.
+	AddTLSPassThroughRoute(routes operatorv1.TLSPassThroughRoute)
+
+	// AddConfigMap accepts a Config Map referenced by a TLS terminated route. This is used to detect changes to ConfigMaps
+	// that will be mounted by the VoltronRouteConfig so Voltron can be restarted if the value changes.
+	AddConfigMap(configMap *corev1.ConfigMap)
+
+	// AddSecret accepts a Secret referenced by a TLS terminated route. This is used to detect changes to Secrets
+	// that will be mounted by the VoltronRouteConfig so Voltron can be restarted if the value changes.
+	AddSecret(secret *corev1.Secret)
+	Build() (*VoltronRouteConfig, error)
+}
+
+type voltronRouteConfigBuilder struct {
+	mountedConfigMaps map[string]struct{}
+	mountedSecrets    map[string]struct{}
+
+	volumeMounts    []corev1.VolumeMount
+	volumes         []corev1.Volume
+	hashableContent []interface{}
+
+	configMaps map[string]*corev1.ConfigMap
+	secrets    map[string]*corev1.Secret
+
+	tlsTerminatedRoutes  []operatorv1.TLSTerminatedRoute
+	tlsPassThroughRoutes []operatorv1.TLSPassThroughRoute
+}
+
+func NewVoltronRouteConfigBuilder() VoltronRouteConfigBuilder {
+	return &voltronRouteConfigBuilder{
+		configMaps:        map[string]*corev1.ConfigMap{},
+		secrets:           map[string]*corev1.Secret{},
+		mountedConfigMaps: map[string]struct{}{},
+		mountedSecrets:    map[string]struct{}{},
+	}
+}
+
+func (builder *voltronRouteConfigBuilder) AddConfigMap(configMap *corev1.ConfigMap) {
+	if _, ok := builder.configMaps[configMap.Name]; !ok {
+		builder.configMaps[configMap.Name] = configMap
+	}
+}
+
+func (builder *voltronRouteConfigBuilder) AddSecret(secret *corev1.Secret) {
+	if _, ok := builder.secrets[secret.Name]; !ok {
+		builder.secrets[secret.Name] = secret
+	}
+}
+
+func (builder *voltronRouteConfigBuilder) AddTLSTerminatedRoute(route operatorv1.TLSTerminatedRoute) {
+	builder.tlsTerminatedRoutes = append(builder.tlsTerminatedRoutes, route)
+}
+
+func (builder *voltronRouteConfigBuilder) AddTLSPassThroughRoute(route operatorv1.TLSPassThroughRoute) {
+	builder.tlsPassThroughRoutes = append(builder.tlsPassThroughRoutes, route)
+}
+
+func (builder *voltronRouteConfigBuilder) Build() (*VoltronRouteConfig, error) {
+	var uiTLSTerminatedRoutes []*tlsTerminatedRoute
```

Comment:
Should we just make this a `[]tlsTerminatedRoute` and get rid of the pointer complexity? Will we ever add `nil` to this slice?

Reply:
This was more so that the reference to route can be updated after it's appended to the array in the loops. We could ensure added at the end of the loop after it's been fully filled out, but I wanted to do the target check first before the other checks which means I'd have to do that check again at the end to find which array to append to.

Seemed simpler just to have this as a pointer, and debatably more intuitive (I was confused why the value wasn't updated in the array when I didn't have this as a pointer).

Reply:
Yeah, I guess it's trading one type of complexity for another. Fine to keep it as is.